### PR TITLE
MINOR: fix core test build

### PR DESCRIPTION
**Pull request by vvcephei**
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by vvcephei** [comment]
Hi @hachikuji , do you mind taking a look at this?

(warning: I ran that one test, but not all the tests, locally before pushing this PR)

File: core/src/test/scala/unit/kafka/log/ProducerStateManagerTest.scala [hunk 1]
```diff
@@ -226,7 +226,12 @@ class ProducerStateManagerTest extends JUnitSuite {
 
     def beginTxn(producerId: Long, startOffset: Long): Unit = {
       val relativeOffset = (startOffset - segmentBaseOffset).toInt
-      val producerAppendInfo = new ProducerAppendInfo(producerId, ProducerStateEntry.empty(producerId), ValidationType.Full)
+      val producerAppendInfo = new ProducerAppendInfo(
+        new TopicPartition("test", 0),
```
Comment (hachikuji) [inline core/src/test/scala/unit/kafka/log/ProducerStateManagerTest.scala]: Can we just use `partition`?

Comment (vvcephei) [reply core/src/test/scala/unit/kafka/log/ProducerStateManagerTest.scala]: oh, sure; I overlooked it.

**Review by hachikuji — APPROVED** [review]
LGTM. Thanks for fixing this. I verified the fix locally, so I will go ahead and merge.

**Pull request merged**